winforms/src/toga_winforms/widgets/table.py

```diff
@@ -173,9 +173,14 @@ def text(attr):
 
         # TODO: ListView only has built-in support for one icon per row. One possible
         # workaround is in https://stackoverflow.com/a/46128593.
-        icon = icon(self._accessors[0])
-        if icon is not None:
-            lvi.ImageIndex = self._image_index(icon)
+        try:
+            icon = icon(self._accessors[0])
+        except IndexError:
```

Comment:
This could hide other IndexErrors from inside `icon()`, so in this case I think it's better to [ask for permission rather than forgiveness](https://stackoverflow.com/a/12270384).